[TRTLLM-1316] refactor: Remove unnecessary pipeline parallelism logic from postProcessRequest

Funatiq opened the pull request:
## Description

- Enable logits validation for multi-gpu executor tests.
- Slightly increase timeout to accommodate for longer generation time.
- Remove unnecessary pipeline parallelism logic from postProcessRequest. Logits don't need to be sent to first pipeline rank because responses are generated on last rank.

## Test Coverage

<!--
Please list clearly what are the relevant test(s) that can safeguard the changes in the PR. This helps us to ensure we have sufficient test coverage for the PR.
-->

## GitHub Bot Help

`/bot [-h] ['run', 'kill', 'skip', 'reuse-pipeline'] ...`

Provide a user friendly way for developers to interact with a Jenkins server.

Run `/bot [-h|--help]` to print this help message.

See details below for each supported subcommand.

<details>

`run  [--disable-fail-fast --skip-test --stage-list "A10-1, xxx" --gpu-type "A30, H100_PCIe" --add-multi-gpu-test --only-multi-gpu-test --disable-multi-gpu-test --post-merge --extra-stage "H100_PCIe-[Post-Merge]-1, xxx"]`

Launch build/test pipelines. All previously running jobs will be killed.

`--disable-fail-fast ` *(OPTIONAL)* : Disable fail fast on build/tests/infra failures.

`--skip-test ` *(OPTIONAL)* : Skip all test stages, but still run build stages, package stages and sanity check stages. Note: Does **NOT** update GitHub check status.

`--stage-list "A10-1, xxx"` *(OPTIONAL)* : Only run the specified test stages. Examples: "A10-1, xxx". Note: Does **NOT** update GitHub check status.

`--gpu-type "A30, H100_PCIe"` *(OPTIONAL)* : Only run the test stages on the specified GPU types. Examples: "A30, H100_PCIe". Note: Does **NOT** update GitHub check status.

`--only-multi-gpu-test ` *(OPTIONAL)* : Only run the multi-GPU tests. Note: Does **NOT** update GitHub check status.

`--disable-multi-gpu-test ` *(OPTIONAL)* : Disable the multi-GPU tests. Note: Does **NOT** update GitHub check status.

`--add-multi-gpu-test ` *(OPTIONAL)* : Force run the multi-GPU tests. Will also run L0 pre-merge pipeline.

`--post-merge ` *(OPTIONAL)* : Run the L0 post-merge pipeline instead of the ordinary L0 pre-merge pipeline.

`--extra-stage "H100_PCIe-[Post-Merge]-1, xxx"` *(OPTIONAL)* : Run the ordinary L0 pre-merge pipeline and specified test stages. Examples: --extra-stage "H100_PCIe-[Post-Merge]-1, xxx".

For guidance on mapping tests to stage names, see `docs/source/reference/ci-overview.md`.

### kill

`kill  `

Kill all running builds associated with pull request.

### skip

`skip --comment COMMENT `

Skip testing for latest commit on pull request. `--comment "Reason for skipping build/test"` is required. IMPORTANT NOTE: This is dangerous since lack of user care and validation can cause top of tree to break.

### reuse-pipeline

`reuse-pipeline `

Reuse a previous pipeline to validate current commit. This action will also kill all currently running builds associated with the pull request. IMPORTANT NOTE: This is dangerous since lack of user care and validation can cause top of tree to break.

</details>

Funatiq committed:
refactor: Improve scripts that generate expected test outputs

Signed-off-by: Robin Kobus <19427718+Funatiq@users.noreply.github.com>

Funatiq committed:
test: Enable logits validation for multi-gpu executor tests

- Updated `runTest` function signatures to include `modelParallelism` parameter.
- Improved validation of context and generation logits with adjustable tolerances.
- Added new file path utilities for expected output files related to model parallelism.
- Adjusted test scripts to generate outputs based on updated parallelism configurations.

Signed-off-by: Robin Kobus <19427718+Funatiq@users.noreply.github.com>

Funatiq committed:
refactor: Remove unnecessary pipeline parallelism logic from postProcessRequest

- Eliminated redundant code related to sending and receiving context and generation logits in pipeline parallelism.
- Logits only need to be available on the last rank in pipeline parallelism to create the request output.

Signed-off-by: Robin Kobus <19427718+Funatiq@users.noreply.github.com>

Funatiq committed:
test: Always return log probs and generation logits in multi-gpu executor test

- Modified test instantiation to set `computeLogProbs` and `returnGenerationLogits` to true, ensuring that both logits are included in the test scenarios.
- This change enhances the coverage of the executor tests by validating the behavior with logits enabled.

Signed-off-by: Robin Kobus <19427718+Funatiq@users.noreply.github.com>

tensorrt-cicd commented:
[PR_Github #9894](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/9894/) [ run ] completed with state `FAILURE`
[/LLM/main/L0_MergeRequest_PR pipeline #7302 `(Partly Tested)`](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/7302/) completed with status: 'FAILURE'

tensorrt-cicd commented:
[PR_Github #10047](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/10047/) [ run ] triggered by Bot

tensorrt-cicd commented:
[PR_Github #10058](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/10058/) [ run ] triggered by Bot

tensorrt-cicd commented:
[PR_Github #10129](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/10129/) [ run ] completed with state `FAILURE`
[/LLM/main/L0_MergeRequest_PR pipeline #7476 `(Partly Tested)`](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/7476/) completed with status: 'FAILURE'

tensorrt-cicd commented:
[PR_Github #10132](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/10132/) [ run ] triggered by Bot

Funatiq commented:
/bot run --stage-list "DGX_H100-4_GPUs-CPP-1"

tensorrt-cicd commented:
[PR_Github #10280](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/10280/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #7598 `(Partly Tested)`](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/7598/) completed with status: 'FAILURE'

tensorrt-cicd commented:
[PR_Github #10298](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/10298/) [ run ] triggered by Bot

Funatiq committed:
feat: Add timing functionality to engine build and output generation scripts

- Introduced timing measurements to `build_llama_engines.py` to log the duration of engine builds.
- Added timing measurements to `generate_expected_llama_output.py` to log the duration of the generation process.

Signed-off-by: Robin Kobus <19427718+Funatiq@users.noreply.github.com>

Funatiq commented:
/bot run --stage-list "DGX_H100-4_GPUs-CPP-1"

tensorrt-cicd commented:
[PR_Github #10392](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/10392/) [ run ] triggered by Bot

tensorrt-cicd commented:
[PR_Github #10392](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/10392/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #7684 `(Partly Tested)`](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/7684/) completed with status: 'FAILURE'

tensorrt-cicd commented:
[PR_Github #10462](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/10462/) [ run ] triggered by Bot

tensorrt-cicd commented:
[PR_Github #10501](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/10501/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #7775](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/7775/) completed with status: 'SUCCESS'

achartier reviewed (approved):
lgtm
